1)Возникновение ошибок при выполнении программы называют исключением. 
try позволяет определить блок кода, который будет проверяться на наличие ошибок во время его выполнения;
throw нужен для создания, отображения исключений и используется для перечисления ошибок;
catch - блок, который выполняется при возникновении определенного исключения в try;
noexcept определяет функцию как не вызывающую исключения и используется в объявлении функции справа от списка параметров;
